src/libsync/owncloudpropagator.h

```diff
@@ -419,6 +461,17 @@ class OwncloudPropagator : public QObject
 
 private slots:
 
+    void abortTimeout()
+    {
+        // Abort synchronously (asyncAbort set to false) and finish
+        QMetaObject::invokeMethod(_rootJob.data(), "abort", Qt::QueuedConnection, Q_ARG(bool, false));
```

Comment:
Is there a reason you use a QueuedConnection here?
the abort(false) will be called after the slotAbortFinished.
If it is required, add a comment why we don't just call abort(false),   if it is not required, just make a call to abort(false)



(also that's a short function, it might be cleaner to use a lambda for it)